### ISSUE TITLE
False positive for unused variable (W0612) on tuple unpacking

### ISSUE THREAD

**Issue by alainlou** [PyCQA/pylint#3781]
### Steps to reproduce
```python3
def function():
    a, b = 0, []

    print(b)

    def inner():
        nonlocal a
        b.append(a)
        a += 1

    return inner()
```

### Current behavior
```
file.py:2:4: W0612: Unused variable 'a' (unused-variable)
```
Interestingly, if you remove the 9th line (`a += 1`), the error goes away.

### Expected behavior
No errors.

### pylint --version output
```
pylint 2.5.2
astroid 2.4.1
Python 3.8.5 (default, Jul 31 2020, 00:00:00) 
[GCC 10.2.1 20200723 (Red Hat 10.2.1-1)]
```

**Comment by AWhetter** [comment]
I can reproduce this. As far as I'm aware, we haven't implemented support for nonlocal yet.
